### PR TITLE
feature(utilities) add automatic DI

### PR DESCRIPTION
**Pull request by AlexTorianyk**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/igroup-opole/yesterday-api/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

**Comment by AlexTorianyk** [comment]
[[backend] Automatic DI](https://trello.com/c/umDsvffj/69-backend-automatic-di)

**Comment by rafalschmidt97** [comment]
What's about that @AlexTorianyk?

screenshot 2018-12-14 at 09 59 08
screenshot 2018-12-14 at 09 59 13

File: YesterdayApi/Utilities/AutomaticDI/AutomaticDI.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+using Microsoft.Extensions.DependencyInjection;
+
+namespace YesterdayApi.Utilities.AutomaticDI
+{
+    public static class AutomaticDI
+    {
+        public static void ConfigureDependencies(this IServiceCollection services)
+        {
+            services.Scan(scan => scan
+                .FromAssemblyOf<IAssemblyMaker>()
```
Comment (rafalschmidt97) [inline YesterdayApi/Utilities/AutomaticDI/AutomaticDI.cs]: IAssemblyMaker is weird but we will improve that in next version

File: YesterdayApi/Utilities/AutomaticDI/IAssemblyMaker.cs [hunk 2]
```diff
@@ -0,0 +1,6 @@
+﻿namespace YesterdayApi.Utilities.AutomaticDI
+{
+    public interface IAssemblyMaker
```
Comment (rafalschmidt97) [inline YesterdayApi/Utilities/AutomaticDI/IAssemblyMaker.cs]: For me it is useless but as @AlexTorianyk said it isn't

File: stylecop.ruleset [hunk 3]
```diff
@@ -8,6 +8,7 @@
     <Rule Id="CA1009" Action="Warning"/>
     <Rule Id="CA1016" Action="Warning"/>
     <Rule Id="CA1033" Action="Warning"/>
+    <Rule Id="CA1040" Action="None"/>
```
Comment (rafalschmidt97) [inline stylecop.ruleset]: I was forced to add this to allow empty interfaces

**Pull request merged**